src/samples/tester.cpp

```diff
@@ -0,0 +1,232 @@
+        case RoomMember::State::MacCollision:
+            std::cout <<  "MAC collision of Member A"<< std::endl;
+            return 1;
+        case RoomMember::State::NameCollision:
+            std::cout <<  "Name Collision of Member A"<< std::endl;
+            return 1;
+        case RoomMember::State::RoomFull:
+            std::cout <<  "Room is to crowded"<< std::endl;
+            return 1;
+        case RoomMember::State::RoomDestroyed:
+            std::cout <<  "Room is dead"<< std::endl;
+            return 1;
+        case RoomMember::State::WrongVersion:
+            std::cout <<  "You got the wrong version"<< std::endl;
+            return 1;
+        case RoomMember::State::Idle:
+            std::cout <<  "This should not happen!"<< std::endl;
+            return 1;
+        default:
+            sleep(1);
+            break;
+        }
+    }
+    std::cout << "Successfully joined Member A" << std::endl;
+    RoomMember::MemberList member_list = member_a->GetMemberInformation();
+    for (const auto member:member_list) {
+        std::ostringstream mac_address;
+        for (const uint8_t& value: member.mac_address) {
+            mac_address << std::hex << std::setw(2) << std::setfill('0') << +value << ":";
+        }
+        std::cout << member.nickname << " " << mac_address.str() << " " << member.game_name << std::endl;
+    }
+
+
+    std::unique_ptr<RoomMember> member_b = std::make_unique<RoomMember>();
+    std::cout << std::endl << "Joining Member B" << std::endl;
+    member_b->Join("Member B");             // Change the ncikname to "Member A" to test name collision
+
+    while ((current_state = member_b->GetState()) != RoomMember::State::Joined) {
+        switch (current_state) {
+        case RoomMember::State::Error:
+            std::cout <<  "Unkown error while connecting Member B"<< std::endl;
+            return 1;
+        case RoomMember::State::LostConnection:
+            std::cout <<  "Lost connection while connecting Member B"<< std::endl;
+            return 1;
+        case RoomMember::State::MacCollision:
+            std::cout <<  "MAC collision of Member B"<< std::endl;
+            return 1;
+        case RoomMember::State::NameCollision:
+            std::cout <<  "Name collision of Member B"<< std::endl;
+            return 1;
+        case RoomMember::State::RoomFull:
+            std::cout <<  "Room is to crowded"<< std::endl;
+            return 1;
+        case RoomMember::State::RoomDestroyed:
+            std::cout <<  "Room is dead"<< std::endl;
+            return 1;
+        case RoomMember::State::WrongVersion:
+            std::cout <<  "You got the wrong version"<< std::endl;
+            return 1;
+        case RoomMember::State::Idle:
+            std::cout <<  "This should not happen!"<< std::endl;
+            return 1;
+        default:
+            sleep(1);
+            break;
+        }
+    }
+    std::cout << "Successfully joined Member B" << std::endl;
```

Comment:
Same block as above = Put it into a function or at least a lambda (But a function is more appropriate here)